packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -426,20 +434,22 @@ class _DatePickerDialogState extends State<DatePickerDialog> with RestorationMix
 
   Size _dialogSize(BuildContext context) {
     final Orientation orientation = MediaQuery.orientationOf(context);
+    final bool useMaterial3 = Theme.of(context).useMaterial3;
```

Comment:
Maybe move this into the switch statements, so that in the case where it's `landscape`, it doesn't need to set up an inheritance dependency on `Theme`? I mean, I know it's probably going to have one anyhow, but if we convert theme to an `InheritedModel` or something someday, that will be important.

Reply:
See https://github.com/flutter/flutter/pull/118603#discussion_r1086008424